examples/Microsoft.Spark.CSharp.Examples/MachineLearning/SentimentStream/README.md

```diff
@@ -0,0 +1,229 @@
+
+```CSharp
+DataFrame words = spark
+    .ReadStream()
+    .Format("socket")
+    .Option("host", hostname)
+    .Option("port", port)
+    .Load();
+```
+
+### 3. Register a UDF to Access ML.NET
+
+A UDF is a *user-defined function.* We can use UDFs in Spark applications to perform calculations and analysis on our data. We create a User Defined Function (UDF) that calls the *Sentiment* method on each piece of text from our stream.
+
+```CSharp
+spark.Udf().Register<string, bool>("MyUDF", input => Sentiment(input));
+```
+
+*Sentiment* is the method where we'll call our ML.NET code that was generated from the final step of Model Builder.
+
+```CSharp
+MLContext mlContext = new MLContext();
+
+ITransformer mlModel = mlContext
+    .Model
+    .Load(modelPath, out var modelInputSchema);
+
+PredictionEngine<Review, ReviewPrediction> predEngine = mlContext
+    .Model
+    .CreatePredictionEngine<Review, ReviewPrediction>(mlModel);
+```
+
+You may notice the use of *Review* and *ReviewPrediction.* These are classes we define in our project to represent the review data we're evaluating. 
+
+```CSharp
+public class Review
+{
+      // Represents the streamed input text
+      [LoadColumn(0)]
+      public string ReviewText;
+}
+```
+
+```CSharp
+public class ReviewPrediction : Review
+{
+
+       [ColumnName("PredictedLabel")]
+       public bool Prediction { get; set; }
+
+       public float Probability { get; set; }
+
+       public float Score { get; set; }
+} 
+```
+
+### 4. Spark SQL
+
+Now that you've read in your data and incorporated ML, use Spark SQL to call the UDF that will run sentiment analysis on each row of your DataFrame:
+
+```CSharp
+DataFrame sqlDf = spark.Sql("SELECT WordsEdit.value, MyUDF(WordsEdit.value) FROM WordsEdit");
+```
+
+### 5. Display Your Stream
+
+We can use `DataFrame.WriteStream()` to establish characteristics of our output, such as printing our results to the console and only displaying the most recent output and not all of our previous output as well. 
+
+```CSharp
+StreamingQuery query = sqlDf
+    .WriteStream()
+    .Format("console")
+    .Start();
+```
+
+## Running Your App
+
+There are a few steps you'll need to follow to build and run your app:
+
+* Start a new netcat server
+* Open a new terminal (separate from the netcat server), move to your app's root folder (i.e. *SentimentStream*)
+* Clean and publish your app
+* Move to your app's `publish` folder
+* `spark-submit` your app from within the `publish` folder
+
+Structured streaming in Spark processes data through a series of small **batches.** 
+When you run your program, the command prompt where we established netcat will allow you to start typing.
+In our example, when you hit *enter* after entering data in the command prompt, Spark will consider that a batch and run the UDF. 
+
+![StreamingOutput](https://github.com/bamurtaugh/spark/blob/StreamingLog/examples/Microsoft.Spark.CSharp.Examples/Sql/Streaming/stream2.png)
```

Comment:
This screenshot doesn't seem to apply to this example.

Reply:
Good catch, removed.